installer/engine_installer/engine_installer.go

```diff
@@ -14,8 +14,227 @@
+}
+
+// ExecHTMLOptions contains fields for html templating in exec.html
+type ExecHTMLOptions struct {
+	BridgeNetwork   template.HTML
+	PublicNetwork   template.HTML
+	ImageStore      template.HTML
+	ComputeResource template.HTML
+	Target          string
+	User            string
+	Password        string
+	Name            string
+	Thumbprint      string
+	CreateCommand   string
+}
+
+func NewEngineInstaller() *EngineInstaller {
+	return &EngineInstaller{Name: "default-vch"}
+}
+
+func (ei *EngineInstaller) populateConfigOptions() *EngineInstallerConfigOptions {
+	defer trace.End(trace.Begin(""))
+
+	vc := ei.validator.IsVC()
+	log.Infof("Is VC: %t\n", vc)
+
+	dcs, err := ei.validator.ListDatacenters()
+	if err != nil {
+		log.Infoln(err)
+		return nil
+	}
+	for _, d := range dcs {
+		log.Infof("DC: %s\n", d)
+	}
+
+	comp, err := ei.validator.ListComputeResource()
+	if err != nil {
+		log.Infoln(err)
+		return nil
+	}
+	for _, c := range comp {
+		log.Infof("compute: %s\n", c)
+	}
+
+	rp, err := ei.validator.ListResourcePool("*")
+	if err != nil {
+		log.Infoln(err)
+		return nil
+	}
+	for _, p := range rp {
+		log.Infof("rp: %s\n", p)
+	}
+
+	nets, err := ei.validator.ListNetworks(!vc) // set to false for vC
+	if err != nil {
+		log.Infoln(err)
+		return nil
+	}
+	for _, n := range nets {
+		log.Infof("net: %s\n", n)
+	}
+
+	dss, err := ei.validator.ListDatastores()
+	if err != nil {
+		log.Infoln(err)
+		return nil
+	}
+	for _, d := range dss {
+		log.Infof("ds: %s\n", d)
+	}
+
+	return &EngineInstallerConfigOptions{
+		Networks:      nets,
+		Datastores:    dss,
+		ResourcePools: rp,
+	}
+}
+
+func (ei *EngineInstaller) buildCreateCommand(binaryPath string) {
+	defer trace.End(trace.Begin(""))
+
+	var createCommand []string
+
+	createCommand = append(createCommand, binaryPath+"/vic/vic-machine-linux")
+	createCommand = append(createCommand, "create")
+	createCommand = append(createCommand, "--no-tlsverify")
+	createCommand = append(createCommand, []string{"--target", ei.Target}...)
+	createCommand = append(createCommand, []string{"--user", ei.User}...)
+	createCommand = append(createCommand, []string{"--password", ei.Password}...)
+	createCommand = append(createCommand, []string{"--name", ei.Name}...)
+	createCommand = append(createCommand, []string{"--public-network", ei.PublicNetwork}...)
+	createCommand = append(createCommand, []string{"--bridge-network", ei.BridgeNetwork}...)
+	createCommand = append(createCommand, []string{"--compute-resource", ei.ComputeResource}...)
+	createCommand = append(createCommand, []string{"--image-store", ei.ImageStore}...)
+	createCommand = append(createCommand, []string{"--thumbprint", ei.Thumbprint}...)
+
+	ei.CreateCommand = createCommand
+}
+
+func (ei *EngineInstaller) verifyLogin() error {
+	defer trace.End(trace.Begin(""))
+
+	ctx := context.TODO()
+
+	var u url.URL
+	u.User = url.UserPassword(ei.User, ei.Password)
+	u.Host = ei.Target
+	u.Path = ""
+	fmt.Printf("server URL: %v\n", u)
```

Comment:
Curious, are we using `fmt` instead of `log` here (and elsewhere) to avoid timestamps on these particular messages?